### PR TITLE
Wizard recipe: opusfile-v0.11.0

### PR DESCRIPTION
**Pull request by aviks**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: opusfile
* Version: v0.11.0

@staticfloat please review and merge.

### CODE REVIEW

File: O/opusfile/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+cd $WORKSPACE/srcdir
+cd opusfile-*
+./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target}
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
```
Comment (giordano) [inline O/opusfile/build_tarballs.jl]: Use `supported_platforms()` and let's see what are the problems?

**Comment by giordano** [comment]
For Windows, [this patch](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-opusfile/no-openssl-wincert.patch) might help

**Comment by aviks** [comment]
FreeBSD fails with 
```
 relocation R_X86_64_PC32 against undefined symbol `__stack_chk_guard@@FBSD_1.0' can not be used when making a shared object; recompile with -fPIC
```
even though i've added `--with-pic` to configure, and I can see that `cc` is using `-fPIC`. Stumped.

**Comment by giordano** [comment]
Yes, but the full error is
```
[11:43:28] /opt/x86_64-unknown-freebsd11.1/bin/x86_64-unknown-freebsd11.1-ld: /opt/x86_64-unknown-freebsd11.1/x86_64-unknown-freebsd11.1/sys-root/usr/lib/libcompat.a(ftime.o): relocation R_X86_64_PC32 against undefined symbol `__stack_chk_guard@@FBSD_1.0' can not be used when making a shared object; recompile with -fPIC
[11:43:28] /opt/x86_64-unknown-freebsd11.1/bin/x86_64-unknown-freebsd11.1-ld: final link failed: Bad value
[11:43:28] clang-9: error: linker command failed with exit code 1 (use -v to see invocation)
```
which means that the object file `ftime.o` in the system archive `/opt/x86_64-unknown-freebsd11.1/x86_64-unknown-freebsd11.1/sys-root/usr/lib/libcompat.a` was not compiled with `-fPIC`.

Following [this answer on StackOverflow](https://stackoverflow.com/a/1351771/2442087) I did
```
sandbox:${WORKSPACE} # cp /opt/x86_64-unknown-freebsd11.1/x86_64-unknown-freebsd11.1/sys-root/usr/lib/libcompat.a .
sandbox:${WORKSPACE} # ar x libcompat.a 
sandbox:${WORKSPACE} # readelf --relocs ftime.o | egrep stack_chk_guard
00000000000f  000400000002 R_X86_64_PC32     0000000000000000 __stack_chk_guard - 4
000000000069  000400000002 R_X86_64_PC32     0000000000000000 __stack_chk_guard - 4
```
So the symbol is marked with `R_X86_64_PC32` -- which we can see also in the error message -- and doesn't match the regexp `(GOT|PLT|JU?MP_SLOT)` suggested in the answer linked above.

I'm not sure what we can do here, probably the safest option for the time being is to not build for FreeBSD.

**Comment by aviks** [comment]
I very happy to not bother with FreeBSD. Will I have to change the `platforms` for all dependees of this?

File: O/opusfile/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+fi
+
+./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} --with-pic
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
```
Comment (giordano) [inline O/opusfile/build_tarballs.jl]:
```suggestion
platforms = filter!(p -> !(p isa FreeBSD), supported_platforms())
```

**Pull request merged**